### DISCUSSION TITLE
Delay showing answer choices to subjects until *after* video finishes?

### DISCUSSION THREAD

**Discussion by BrianOdegaard** [jspsych/jsPsych#944] [Q&A] [answered]
I'm conducting a study where I will show videos to subjects using the jspsych-video-keyboard-response plugin.  After watching videos, subjects will have to report the identity of the stimulus in the video.

The answer choices that will be shown to subjects are contingent on which video is shown, and I have been able to incorporate that contingency using information found in the jsPsych Timeline Variables tutorial:

https://www.jspsych.org/overview/timeline/#timeline-variables

However, it would be nice for these answer choices to be displayed *after* the video finishes playing.  Is there any way to do this with the current plugin?  If not, do you have a recommendation for what the best hack would be to pull it off?

Thx,
Brian

**Comment by vijaymarupudi** [comment, marked as answer]
Hello Brian,

Would adding an appropriate trial (survey-html-form, survey-text, etc.) after the video trial achieve the behavior you want?

**Reply by BrianOdegaard** [reply]
Probably.  I need to do a little bit of work on this - since the answer choices are contingent on which specific video is shown, and I'm using timeline variables, I'm not sure how to add the extra trial after the video and ensure that the correct answer choices are shown, but I'll work through it and follow up if I have questions.

**Reply by vijaymarupudi** [reply]
Sure! What's key here are dynamic parameters: https://www.jspsych.org/overview/trial/#dynamic-parameters

Use a function for the form/questions parameters and then you can extract the video name from the previous trial and condition on that for the questions shown to the participants.

**Reply by BrianOdegaard** [reply]
Awesome - thank you, Vijay!  This project is my first attempt to actually use timeline variables properly ( I used to just use for loops, etc. to set up my experiments), so I'll take a look at dynamic parameters and see if I can properly extract info about the last trial.  Much appreciated!
